File: class/Wheel.py
###Roue###
#-pneu#
#-diamètre#
#-périmètre#

class Wheel:
    def __init__(self, tireReference):
        self.tireReference = tireReference
        self.tireWidth = int(self.tireReference[0] + self.tireReference[1] + self.tireReference[2])
        self.tirePercent = int(self.tireReference[4] + self.tireReference[5])
        self.tireSize = int(self.tireReference[8] + self.tireReference[9])
        self.diameter = (self.tireWidth*(self.tirePercent/100)*2+self.tireSize*25.4)/1000
        self.perimeter = 3.142*(self.diameter/2)**2
        
    def update(self, carSpeed):
        self.rpm = carSpeed/self.perimeter
    
    def print(self):
        print(self.tireReference)
        print(self.tireWidth)
        print(self.tirePercent)
        print(self.tireSize)
        print(round(self.diameter, 3))

File: libs/src/Wheel.cpp
#include "../../include/Wheel.hpp"

Wheel::Wheel(char* TireReference){
  tireReference = TireReference;
  tireWidth = 185;
  tirePercent = 60;
  tireSize = 14;
  diameter = (tireWidth*(tirePercent/100)*2+tireSize*25.4)/1000;
  perimeter = 3.142*(diameter/2)*2;
  loger.construct(">>Wheel Construction<<");
}

Wheel::~Wheel(){
  loger.destruct("##Wheel Destruction##");
}

void Wheel::update(float carSpeed){
  rpm = carSpeed/perimeter;
}

void Wheel::print(){
  //loger.log((char*)tireWidth);
}
